Dart/src/com/jetbrains/lang/dart/ide/generation/BaseDartGenerateAction.java

```diff
@@ -39,9 +39,26 @@ public void update(final AnActionEvent e) {
     final PsiFile psiFile = editorAndPsiFile.second;
 
     final int caretOffset = editor == null ? -1 : editor.getCaretModel().getOffset();
-    final boolean inClass = psiFile != null && PsiTreeUtil.getParentOfType(psiFile.findElementAt(caretOffset), DartClass.class) != null;
+    final boolean enable = psiFile != null && doEnable(PsiTreeUtil.getParentOfType(psiFile.findElementAt(caretOffset), DartClass.class));
 
-    e.getPresentation().setEnabled(inClass);
-    e.getPresentation().setVisible(inClass);
+    e.getPresentation().setEnabled(enable);
+    e.getPresentation().setVisible(enable);
   }
+
+  protected boolean doEnable(@Nullable final DartClass dartClass) {
+    return dartClass != null;
+  }
+
+  protected static boolean doesClassContainMember(@NotNull final DartClass dartClass, @NotNull final String memberName) {
+    if (memberName.isEmpty()) {
+      return false;
+    }
+    final DartComponent dartComponent = dartClass.findMemberByName(memberName);
+    if (dartComponent == null) {
+      return true;
+    }
+    final DartClass containingClass = PsiTreeUtil.getParentOfType(dartComponent, DartClass.class);
+    return containingClass != dartClass;
```

Comment:
As we do not need to iterate supeclasses here we can use `DartResolveUtil#getNamedSubComponents` that looks into current class only.
Or even more optimal: iterate `PsiTreeUtil.getChildrenOfType(DartResolveUtil.getBody(dartClass), DartMethodDeclaration.class)` checking for `toString()` among them.